libraries/SITL/examples/JSON/C++/Socket.h

```diff
@@ -0,0 +1,76 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/*
+  simple socket handling class for systems with BSD socket API
+ */
+#pragma once
+
+// #include <AP_HAL/AP_HAL.h>
+// #if HAL_OS_SOCKETS
+
+#include <cstring>
+
+#include <fcntl.h>
+#include <unistd.h>
+#include <sys/ioctl.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <netinet/tcp.h>
+#include <arpa/inet.h>
+#include <sys/select.h>
+
+class SocketAPM {
```

Comment:
Rename

Move to C++ code as being an internal thing?

Reply:
Renamed to SocketExample